src/tests/ftest/util/results_utils.py

```diff
@@ -0,0 +1,331 @@
+"""
+  (C) Copyright 2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+
+import time
+
+
+class TestName():
+    # pylint: disable=too-few-public-methods
+    """Provides the necessary test name data to generate a xml/html results file."""
+
+    def __init__(self, name, uid, variant):
+        """Initialize a TestName object.
+
+        Args:
+            name (str): the test name
+            uid (str): the test uid
+            variant (str): the test variant
+        """
+        self.name = name
+        self.uid = uid
+        self.variant = variant
+
+    def __str__(self):
+        """Get the test name as a string.
+
+        Returns:
+            str: the test name including the uid, name, and variant
+
+        """
+        return "-".join([str(self.uid), "".join([str(self.name), str(self.variant)])])
```

Comment:
Nitpick - I feel like this is more readable
```suggestion
        return "{}-{}{}".format(self.uid, self.name, self.variant)
```

Reply:
Done